### PR TITLE
ALlow us to take the old choose peer logic back

### PR DESCRIPTION
**Pull request by Raynos**
I also fixed a few tests.

r: @kriskowal @shannili

### CODE REVIEW

**Comment by ShanniLi** [comment]
lgtm. It would be nice to provide an interface to update chooseHeapPeers during runtime.

**Comment by Raynos** [comment]
@ShanniLi agreed; we need that interface, in the future :)

File: node/channel.js [hunk 1]
```diff
@@ -147,6 +147,7 @@ function TChannel(options) {
     self.emitConnectionMetrics =
         typeof self.options.emitConnectionMetrics === 'boolean' ?
         self.options.emitConnectionMetrics : true;
+    self.chooseHeapPeers = self.options.chooseHeapPeers || false;
```
Comment (kriskowal) [inline node/channel.js]: useHeap, choosePeerWithHeap

File: node/out_request.js [hunk 2]
```diff
@@ -83,7 +83,7 @@ function TChannelOutRequest(id, options) {
         self.setupTracing(options);
     }
 
-    self.peer.invalidateScore();
+    // self.peer.invalidateScore();
```
Comment (kriskowal) [inline node/out_request.js]: deliberate?

Comment (Raynos) [inline node/out_request.js]: yes.

Comment (kriskowal) [inline node/out_request.js]: Perhaps explain or delete the line. Maybe leave comment on why.

Comment (Raynos) [inline node/out_request.js]: delete line; will fix.

File: node/test/retry.js [hunk 3]
```diff
@@ -99,35 +78,33 @@ allocCluster.test('request retries', {
...
-            req.outReqs[0] &&
-            req.outReqs[0].err &&
-            req.outReqs[0].err.type,
-            'tchannel.declined',
-            'expected first request to decline');
+        var reqErrors = req.outReqs.filter(function isErr(r) {
+            return r.err;
+        });
+        var reqResp = req.outReqs.filter(function isResp(r) {
+            return r.res;
```
Comment (kriskowal) [inline node/test/retry.js]: isRes, hasRes. Let’s not make new abbreviations for response.

File: node/test/retry.js [hunk 4]
```diff
@@ -99,35 +78,33 @@ allocCluster.test('request retries', {
...
+                'expected error request'
+            );
+        }
+
+        assert.ok(reqResp[0] &&
+                  reqResp[0].res, 'expected to have 4th response');
+        assert.deepEqual(reqResp[0] &&
+                         reqResp[0].res.arg2, arg2, 'arg2 came form 4th response');
+        assert.deepEqual(reqResp[0] &&
+                         reqResp[0].res.arg3, arg3, 'arg3 came form 4th response');
```
Comment (kriskowal) [inline node/test/retry.js]: Although, @jcorbin will likely shed a tear, this does seem less fragile going forward, as we test the intended behavior and not implementation details that alternate algorithms might not exhibit.

**Comment by kriskowal** [comment]
:ok:

**Pull request merged**